### DISCUSSION TITLE
How to properly imperatively set queryData?

### DISCUSSION THREAD

**Discussion by Jurek-Raben** [TanStack/query#7195] [Q&A] [answered]
Hi,

I would like to manually set the data of a query cache, then route to another page and there use useQuery to get that just set data...

So this is what I am doing:

On first page:
```
await queryClient.setQueryData('cache-key', dataObject);
navigate(nextPage);

```
On the next page:

```
function Component() {
  const { data } = useQuery( {
    queryKey: ['cache-key'],
    queryFn: async () => {
      return null; // dummy return
    },
  });
...
```

But data always is null.

What am I doing wrong here? :)

**Comment by TkDodo** [comment]
likely, the `queryFn` runs because of default staleTime zero and overrides your imperatively set value ?

**Comment by Jurek-Raben** [comment, marked as answer]
Thanks for the hint, staleTime actually was set to Infinity, but I think I simply missed the "await", also changed it to setQueryData(['cache-key'], dataObject) and now it works!
